workflow/kfdrc_production_amplicon_architect.cwl

```diff
@@ -0,0 +1,286 @@
+        required: false}]}
+  annotation_file: {type: 'File?', doc: "refFlat.txt file, needed if cnv kit cnn not\
+      \ already built"}
+  male_input_flag: {type: 'boolean?', doc: "Is the input male?", default: false}
+
+outputs:
+  aa_cnv_seeds: {type: File, doc: "Bed file with candidate regions to search", outputSource: prepare_aa/aa_cnv_seeds}
+  aa_summary: {type: File, doc: "summary for all amplicons detected by AA", outputSource: amplicon_architect/summary}
+  aa_cycles: {type: 'File[]', doc: "text file for each amplicon listing the edges\
+      \ in the breakpoint graph, their categorization (sequence, discordant, concordant,\
+      \ source) and their copy counts", outputSource: amplicon_architect/cycles}
+  graph: {type: 'File[]', doc: 'A text file for each amplicon listing the edges in
+      the breakpoint graph, their categorization (sequence, discordant, concordant,
+      source) and their copy counts', outputSource: amplicon_architect/graph}
+  sv_pdf: {type: 'File[]', doc: "PDF image file displaying the SV view of AA", outputSource: amplicon_architect/sv_pdf}
+  sv_png: {type: 'File[]', doc: "PNG image file displaying the SV view of AA", outputSource: amplicon_architect/sv_png}
+  amplicon_classification_profiles: {type: 'File[]?', doc: "abstract classification\
+      \ of the amplicon", outputSource: amplicon_classifier/amplicon_classification_profiles}
+  gene_list: {type: 'File[]?', doc: "genes present on amplicons with each classification",
+    outputSource: amplicon_classifier/gene_list}
+  ecDNA_cts: {type: 'File[]?', doc: "not yet defined", outputSource: amplicon_classifier/ecDNA_cts}
+
+steps:
+  samtools_tumor_cram_to_bam:
+    run: ../tools/samtools_calmd.cwl
+    when: $(inputs.input_reads.nameext == ".cram")
+    in:
+      input_reads: tumor_align_file
+      reference: reference
+    out: [bam_file]
+
+  samtools_normal_cram_to_bam:
+    run: ../tools/samtools_calmd.cwl
+    when: $(inputs.input_reads != null && inputs.input_reads.nameext == ".cram")
+    in:
+      input_reads: normal_align_file
+      reference: reference
+    out: [bam_file]
+
+  untar_data_repo:
+    run: ../tools/untar_gzip.cwl
+    in:
+      input_tar: aa_data_repo
+      out_dir_name:
+        valueFrom: ${return "data_repo"}
+    out: [output]
+
+
+  cnvkit_batch:
+    run: ../tools/cnvkit_batch_only.cwl
+    when: $(inputs.cnvkit_cnn != null || inputs.input_control != null)
+    in:
+      input_sample:
+        source: [samtools_tumor_cram_to_bam/bam_file, tumor_align_file]
+        pickValue: first_non_null
+      input_control:
+        source: [samtools_normal_cram_to_bam/bam_file, normal_align_file]
+        pickValue: first_non_null
+      reference:
+        # samtools needs fasta, can't use fasta if cnn file provided
+        source: [cnvkit_cnn, reference]
+        valueFrom: "${ if (self[0] == null){ return self[1]; } else{ return null;\
+          \ } }"
```

Comment:
I'm pretty sure you can use the `inputs` object here. According to the [spec](https://www.commonwl.org/v1.2/Workflow.html#WorkflowStepInput):
> The value of `inputs` in the parameter reference or expression must be the input object to the workflow step after assigning the `source` values, applying `default`, and then scattering.

Since you are passing in `cnvkit_cnn` below you can set this to:  
```
source: reference
valueFrom: $(inputs.cnvkit_cnn == null ? self : null)
```
Also, the ternary operator is your friend: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator